### PR TITLE
Update Enums.jl

### LINKED ISSUE #28778: Convert Enum to integer value with correct type
Hello,

Following https://github.com/andrewadare/LibSerialPort.jl/pull/19#issuecomment-414151579 and discussion with @tanmaykm and @andrewadare maybe Enum docstring should be improved to show how to get BaseType.

According https://github.com/andrewadare/LibSerialPort.jl/pull/19#issuecomment-414190281 it can be done using

```julia
julia> @enum Fruit::UInt8 apple orange kiwi

julia> Base.Enums.basetype(Fruit)
UInt8
```

but we should also document how to get integer value of an Enum with correct type, I have found

```julia
julia> Base.Enums.basetype(typeof(kiwi))(kiwi)
0x02
```

but maybe there is a simpler solution which should, in such a case, be documented because in most case it's suggested to use `Int`, but it's changing type

```julia
julia> supertype(Fruit)
Enum{UInt8}

julia> typeof(Int(apple))
Int64
```

Kind regards

### PR DESCRIPTION
**Pull request by s-celles**
Closes #28778

### CODE REVIEW

File: base/Enums.jl [hunk 1]
```diff
@@ -68,6 +68,38 @@ To list all the instances of an enum use `instances`, e.g.
...
+```julia
+julia> @enum Vegetable::UInt8 pea carrot broccoli
+
+julia> Vegetable
+Enum Vegetable:
+pea = 0x00
+carrot = 0x01
+broccoli = 0x02
+
+julia> Base.Enums.basetype(Vegetable)
```
Comment (KristofferC) [inline base/Enums.jl]: Can't we just use the enum already defined?

Comment (scls19fr) [reply base/Enums.jl]: No because that's for showing a `BaseType` which is different from default one.

Comment (KristofferC) [reply base/Enums.jl]: This docstring is becoming very long so we need to think about ways of shortening it. I think adding a docstring to `Enums.basetype` and just referring to it here is a good idea?

File: base/Enums.jl [hunk 2]
```diff
@@ -62,6 +94,8 @@ end
 Member values can be converted between the enum type and `BaseType`. `read` and `write`
 perform these conversions automatically.
 
+`Base.Enums.basetype` function returns `BaseType`.
```
Comment (KristofferC) [inline base/Enums.jl]: We probably want `[Base.Enums.basetype](@ref)` and also have the docstring for `basetype` show up in the list of docstrings in the online manual by adding it to the list here https://github.com/JuliaLang/julia/blob/master/doc/src/base/base.md#special-types.

**Comment by scls19fr** [comment]
That's done @KristofferC

**Comment by ViralBShah** [comment]
Do we want to get this in?

**Comment by ViralBShah** [comment]
Superseded by #36057

**Pull request closed without merging**